File: vba/euler1.vb
Function euler1(x)
    euler1 = 0

    For i = 1 To x - 1
        If i Mod 3 = 0 Or i Mod 5 = 0 Then
            euler1 = euler1 + i
        End If
    Next
End Function

Sub main()
    MsgBox (euler1(1000))
End Sub

File: standard_ml/euler1.ml
(* Euler1 in Standard ML *) 

local
    fun f 1 acc = acc
      | f i acc = f (i-1) (if i mod 3 = 0 orelse i mod 5 = 0
                            then acc + i
                            else acc)
in
    fun euler1 size = f size 0
end; 

print (Int.toString( euler1 999 ) ^ "\n");